This patch chagnes the way that BUCK1 and BUCK2 of rk80x PMIC set voltage.

It changes from making max. 100mV per step into max. 50mV per step.

This means that for overclocked LiTTLE cores with base 408MHz @ 0.825V
and max. 1.5GHz @ 1.2V it will take 7 steps of 50mV (at least 65uS each),
the final 25mV step and 50uS to settle: 7 x 65uS + 50uS = 505uS. 

Before the change it would be 3 steps of 100mV (at least 65uS each),
the final 75mV step and 1uS to settle: 3 x 65uS + 1uS = 196uS.

This is 2.5x longer but it makes NanoPi M4V2 stable again.

Signed-off-by: Piotr Szczepanik <piter75@gmail.com>

diff --git a/drivers/regulator/rk808-regulator.c b/drivers/regulator/rk808-regulator.c
index e926c1a85..e9bb23683 100644
--- a/drivers/regulator/rk808-regulator.c
+++ b/drivers/regulator/rk808-regulator.c
@@ -61,7 +61,7 @@
 #define RK808_SLP_SET_OFF_REG_OFFSET 2
 
 /* max steps for increase voltage of Buck1/2, equal 100mv*/
-#define MAX_STEPS_ONE_TIME 8
+#define MAX_STEPS_ONE_TIME 4
 
 #define ENABLE_MASK(id)			(BIT(id) | BIT(4 + (id)))
 #define DISABLE_VAL(id)			(BIT(4 + (id)))
@@ -282,7 +282,7 @@ static int rk808_buck1_2_i2c_set_voltage_sel(struct regulator_dev *rdev,
 	 * 100000uv/us, wait 1us to make sure the target voltage to be stable,
 	 * so we needn't wait extra time after that.
 	 */
-	udelay(1);
+	udelay(50);
 
 	return ret;
 }
